examples/timeseries/timeseries_transformer_classification.py

```diff
@@ -0,0 +1,189 @@
+    data = np.loadtxt(filename, delimiter="\t")
+    y = data[:, 0]
+    x = data[:, 1:]
+    return x, y.astype(int)
+
+root_url = "https://raw.githubusercontent.com/hfawaz/cd-diagram/master/FordA/"
+
+x_train, y_train = readucr(root_url + "FordA_TRAIN.tsv")
+x_test, y_test = readucr(root_url + "FordA_TEST.tsv")
+
+x_train = x_train.reshape((x_train.shape[0], x_train.shape[1], 1))
+x_test = x_test.reshape((x_test.shape[0], x_test.shape[1], 1))
+
+n_classes = len(np.unique(y_train))
+
+idx = np.random.permutation(len(x_train))
+x_train = x_train[idx]
+y_train = y_train[idx]
+
+y_train[y_train == -1] = 0
+y_test[y_test == -1] = 0
+
+"""
+## Model Creation
+
+The dataset is now ready and moving on, we are going to implement our model, which first encodes a tensor of shape `(batch size, sequence length, features)`, where `sequence length` is the number of time steps and `features` is each input time series. This is identical to the LSTM layer that keras provides.
+"""
+
+import tensorflow.keras as keras
+import tensorflow.keras.backend as K
+
+"""
+Here, we make the MultiHeadAttention layer to perform
+self-attention, by generating each of {query, key, value}
+matrices, through a dense layer.
+This corresponds to implementing an attention mechanism by:
+`Q = dense_q(x), K = dense_k(x), V = dense_v(x)` , where
+each `dense_x()` is a different instance of `keras.layers.Dense`.
+
+(The actual implementation is a much more optimized version.)
+"""
+
+class SelfAttention(keras.Model):
+    def __init__(self, head_size, num_heads=1, dropout=0, name='SelfAttention', **kwargs):
+        super().__init__(name=name, **kwargs)
+        self.attention = keras.layers.MultiHeadAttention(
+            num_heads=num_heads, key_dim=head_size, dropout=dropout, **kwargs)
+
+    # input shape = output shape = (batch size, seq len, features)
+    def call(self, inputs, **kwargs):
+        x = inputs
+        return self.attention(query=x, key=x, value=x, **kwargs)
+
+    def compute_output_shape(self, input_shape):
+        return input_shape
+
+"""
+The Self-Attention mechanism is weak on it's own. We need to add a residual connection, (layer) normalization,
+dropout and projection layers for this to work. The following layer can also be stacked multiple times.
+
+The projection layers are implemented through `keras.layers.Conv1D`, by (automatically) broadcasting the compute to the last feature dimension of the input tensors.
+"""
+
+class SelfAttentionBlock(keras.Model):
+    def __init__(self,head_size, num_heads=1, ff_dim=None, dropout=0, name='SelfAttentionBlock',  **kwargs):
+        super().__init__(name=name, **kwargs)
+
+        if ff_dim is None:
+            ff_dim = head_size
+
+        self.attention = SelfAttention(head_size, num_heads, dropout=dropout)
+        self.attention_dropout = keras.layers.Dropout(dropout)
+        self.attention_norm = keras.layers.LayerNormalization(epsilon=1e-6)
+
+        self.ff_conv1 = keras.layers.Conv1D(
```

Comment:
It would be more general to use a Sequential for this instead of 2 separate layers defined in 2 different places